__tests__/MongoBulkDataMigration.rollback.test.ts

```diff
@@ -882,5 +882,67 @@ describe('MongoBulkDataMigration', () => {
         ]);
       });
     });
+
+    describe('arrayFilters support', () => {
```

Comment:
@pp0rtal Sorry to ruin the day, but when a value is already set, the rollback will fail as the existing value disappears.

Here is a sample test:

```ts
it('failing tests', async () => {
        const matchDocument = {
          keys: [
            {
              subKey: 'match_me',
              value: "I am already set",
            },
          ],
        };
        const insertResult = await collection.insertMany([
          matchDocument,
        ]);
        const insertedDocuments = await collection
          .find({ _id: { $in: Object.values(insertResult.insertedIds) } })
          .toArray();

        const migration = new MongoBulkDataMigration<any>({
          ...DM_DEFAULT_SETUP,
          update: {
            $set: {
              'keys.$[element].value': '____MATCHED____',
            },
          },
          options: {
            arrayFilters: [
              { 'element.subKey': 'match_me' },
            ],
            projectionBackupFilter: [],
          },
        });
```

Reply:
Nevermind! Removing `projectionBackupFilter` does the trick 😅

Reply:
@hugop95 Sorry to have made you wasted time on this, cases were not covered enough. I have added a test case for `$set` (here we just needed `$unset`) acbe9b7976c33d9c7037a742e3b5bd1194cc955a

> Would it be possible to handle should rollback automatically change on nested keys without having to set projectionBackupFilter: [] ?

I don't manage to find an easy solution. I wonder if it will be easy as we are mixing both positional and static offset. I just opened an issue to keep track of this https://github.com/360Learning/mongo-bulk-data-migration/issues/24

For a first addition to `arrayFilters`, this rollback is a pure bonus so I think we can postpone the next brain session